### PR TITLE
Cleanup item::get_remaining_capacity()

### PR DESCRIPTION
**Pull request by hesu**
Remove `item::get_remaining_capacity()` and replace it to `item::get_remaining_capacity_for_liquid()`. 
- Because `item::get_remaining_capacity()` functions very similar to `..._for_liquid()`, and all callers use the function for checking liquids.

`recipe::check_eligible_containers_for_crafting` tweaks.
- Sort first, so don't loop twice.

### CODE REVIEW

File: src/crafting.cpp [hunk 1]
```diff
@@ -378,35 +378,26 @@ bool recipe::check_eligible_containers_for_crafting(int batch) const
     all.insert(all.end(), res.begin(), res.end());
     all.insert(all.end(), bps.begin(), bps.end());
 
-    for(item &prod : all) {
+    for(item & prod : all) {
         if (prod.made_of(LIQUID)) {
```
Comment (bkentel) [inline src/crafting.cpp]: You could flip this to the opposite condition and `continue` to reduce the indentation level a bit.

File: src/crafting.cpp [hunk 2]
```diff
@@ -378,35 +378,26 @@ bool recipe::check_eligible_containers_for_crafting(int batch) const
...
-            // we go trough half-filled containers first
-            for(item &cont : conts) {
-                if (!cont.is_container_empty()) {
-                    if (cont.contents[0].type->id ==  prod.type->id) {
-                        charges_to_store -= cont.get_remaining_capacity();
-                        if (charges_to_store <= 0) {
-                            break;
-                        }
-                    }
+            for( item & cont : conts) {
```
Comment (bkentel) [inline src/crafting.cpp]: `cont` could be const, no?

File: src/item.cpp [hunk 3]
```diff
@@ -4252,6 +4239,11 @@ bool item_compare_by_charges( const item *left, const item *right)
     }
 }
 
+bool item_compare_by_charges( const item left, const item right)
```
Comment (bkentel) [inline src/item.cpp]: `left` and `right` should be taken by const ref, not value, no?

**Comment by hesu** [comment]
Thank you. Applied the advice.

**Pull request merged**